File: go/2154.go
func findFinalValue(nums []int, original int) int {
    
    values := make(map[int]bool)
    for _, num := range nums {
        values[num] = true
    }

    copy := original
    for {
        if _, ok := values[copy]; ok {
            copy *= 2
            continue
        }
        break
    }
    return copy
}

File: rust/2154.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_final_value(nums: Vec<i32>, original: i32) -> i32 {
        
        let mut copy = original;
        let values: HashSet<i32> = nums.into_iter().collect();

        loop {
            if values.contains(&copy) {
                copy *= 2;
                continue;
            }
            break;
        }

        copy
    }
}
